### ISSUE TITLE
kind fails at "Joining worker nodes"

### ISSUE THREAD

**Issue by vinibali** [kubernetes-sigs/kind#3349]
**What happened**:
Cluster creation failed on joining worker nodes

**What you expected to happen**:
Cluster creation ended successfully after joining worker nodes

**How to reproduce it (as _minimally_ and precisely as possible)**:
kind create cluster --name local-cluster --config $HOME/bin/kind-example-config.yaml --retain
```
# this config file contains all config fields with comments
# NOTE: this is not a particularly useful config file
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
# patch the generated kubeadm config with some extra settings
kubeadmConfigPatches:
- |
  apiVersion: kubelet.config.k8s.io/v1beta1
  kind: KubeletConfiguration
  evictionHard:
    nodefs.available: "0%"
# patch it further using a JSON 6902 patch
kubeadmConfigPatchesJSON6902:
- group: kubeadm.k8s.io
  version: v1beta3
  kind: ClusterConfiguration
  patch: |
    - op: add
      path: /apiServer/certSANs/-
      value: my-hostname
# 1 control plane node and 3 workers
nodes:
# the control plane node config
- role: control-plane
# the three workers
- role: worker
```

**Anything else we need to know?**:
Fresh installation with Ubuntu 22.04, some days before the 20.04 was working fine.

**Environment:**

- kind version: kind v0.18.0 go1.20.2 linux/amd64
- Runtime info:
```
$ docker info 
Client:       
 Version:    24.0.5  
 Context:    default
 Debug Mode: false     
                                           
Server:                                  
 Containers: 2                   
  Running: 0      
  Paused: 0         
  Stopped: 2         
 Images: 2   
 Server Version: 24.0.5     
 Storage Driver: overlay2
  Backing Filesystem: btrfs                                                           
  Supports d_type: true                                                               
  Using metacopy: false
  Native Overlay Diff: true                                                           
  userxattr: false                                                                    
 Logging Driver: json-file                                                            
 Cgroup Driver: systemd                    
 Cgroup Version: 2                                                                    
 Plugins:                 
  Volume: local                   
  Network: bridge host ipvlan macvlan null overlay                                    
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive             
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc                                                                
 Init Binary: docker-init                                                             
 containerd version:                   
 runc version:                                                                        
 init version:                                                                        
 Security Options:                                                                    
  apparmor                                                                            
  seccomp                                                                             
   Profile: builtin                                                                   
  cgroupns                             
 Kernel Version: 6.2.0-31-generic
 Operating System: Ubuntu 22.04.3 LTS                                                 
 OSType: linux
 Architecture: x86_64
 CPUs: 8            
 Total Memory: 31.03GiB
 Name: elx-5cg1467389                      
 ID: 3fd23646-0ef4-44da-9494-f938028e02c6
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false
```
- OS (e.g. from `/etc/os-release`): Ubuntu LTS 22.04
- Kubernetes version: 1.28.1
- Any proxies or other special environment settings?: GlobalProtect VPN in place

**Comment by vinibali** [comment]
[kind_log.tar.gz](https://github.com/kubernetes-sigs/kind/files/12504096/kind_log.tar.gz)

**Comment by stmcginnis** [comment]
Can you include the `docker info` output requested by the issue template? That usually has a lot of very useful details that can help narrow down where a failure is coming from.

**Comment by aojea** [comment]
> GlobalProtect VPN in place

Does it work without the VPN?

**Comment by vinibali** [comment]
> Can you include the `docker info` output requested by the issue template? That usually has a lot of very useful details that can help narrow down where a failure is coming from.

added, thanks

**Comment by vinibali** [comment]
> > GlobalProtect VPN in place
> 
> Does it work without the VPN?

nope

**Comment by BenTheElder** [comment]
for the future:

please find the minimum reproducer of this failure, e.g. does it fail on your hosts with a single node? with two nodes and no other config options? or only once you add the additional options?

> kind version: kind v0.18.0 go1.20.2 linux/amd64

for this bug: please upgrade to a current release. I skimmed your logs and I see an issue pulling the kindnetd image behind the vpn.

the kindnetd image should be preloaded and not need pulling. this bug was fixed in v0.19 https://github.com/kubernetes-sigs/kind/releases/tag/v0.19.0#fixes

**Comment by vinibali** [comment]
> for the future:
> 
> please find the minimum reproducer of this failure, e.g. does it fail on your hosts with a single node? with two nodes and no other config options? or only once you add the additional options?
> 
> > kind version: kind v0.18.0 go1.20.2 linux/amd64
> 
> for this bug: please upgrade to a current release. I skimmed your logs and I see an issue pulling the kindnetd image behind the vpn.
> 
> the kindnetd image should be preloaded and not need pulling. this bug was fixed in v0.19 https://github.com/kubernetes-sigs/kind/releases/tag/v0.19.0#fixes

I'll take a look later this week and do a feedback. Thank you!
